kuka-driver/kuka_driver.cc

```diff
@@ -106,131 +203,114 @@ class KukaLCMClient : public KUKA::FRI::LBRClient {
 
   virtual void monitor() {
     KUKA::FRI::LBRClient::monitor();
-    PublishStateUpdate();
-  }
+    lcm_client_->UpdateRobotState(robot_id_, robotState());
+ }
 
   virtual void waitForCommand() {
     KUKA::FRI::LBRClient::waitForCommand();
 
     // The value of the torques sent in waitForCommand doesn't matter,
     // but we have to send something.
     if (robotState().getClientCommandMode() == KUKA::FRI::TORQUE) {
-      double torque[num_joints_] = { 0., 0., 0., 0., 0., 0., 0.};
+      double torque[kNumJoints] = { 0., 0., 0., 0., 0., 0., 0.};
       robotCommand().setTorque(torque);
     }
 
-    PublishStateUpdate();
+    lcm_client_->UpdateRobotState(robot_id_, robotState());
```

Comment:
why are we doing querying state when waiting for command? 

since you are polling command (from lcm) in the main loop, there really isn't anything to do in the waitForCommand part?